core/src/wsv.rs

```diff
@@ -279,23 +279,155 @@ impl WorldStateView {
+                                        return;
+                                    }
+                                }
+
+                                match self.process_trigger(action, event_type) {
+                                    Ok(()) => succeed.push(id),
+                                    Err(err) => errors.push(err),
+                                }
+                            })
+                        }
+                        Event::Time(_) => {
+                            let dangerous_reference = unsafe {
+                                let ptr = &self.world.triggers as *const TriggerSet; // TODO: REALLY FIX THIS
+                                &*ptr
+                            };
+
+                            dangerous_reference.time_triggers.get(&id).map(|entry| {
+                                let action = entry.value();
+                                if let Repeats::Exactly(atomic) = action.repeats() {
+                                    if atomic.get() == 0 {
+                                        return;
+                                    }
+                                }
+
+                                match self.process_trigger(action, event_type) {
+                                    Ok(()) => succeed.push(id),
+                                    Err(err) => errors.push(err),
+                                }
+                            })
+                        }
+                        Event::ExecuteTrigger(_) => {
+                            let dangerous_reference = unsafe {
+                                let ptr = &self.world.triggers as *const TriggerSet; // TODO: REALLY FIX THIS
+                                &*ptr
+                            };
+
+                            dangerous_reference.by_call_triggers.get(&id).map(|entry| {
+                                let action = entry.value();
+                                if let Repeats::Exactly(atomic) = action.repeats() {
+                                    if atomic.get() == 0 {
+                                        return;
+                                    }
+                                }
+
+                                match self.process_trigger(action, event_type) {
+                                    Ok(()) => succeed.push(id),
+                                    Err(err) => errors.push(err),
+                                }
+                            })
+                        }
+                    };
+                }
+
+                if errors.is_empty() {
+                    (succeed, Ok(()))
+                } else {
+                    (succeed, Err(errors))
+                }
+            };
+
+            for id in &succeed {
+                // Ignoring error if trigger has not `Repeats::Exact(_)` but something else
+                let _mod_repeats_res = self.world.triggers.mod_repeats(id, |n| {
+                    if n == 0 {
```

Comment:
Rust is expression based, this if can be used as a ternary operator.

Reply:
sure